src/main/java/org/assertj/core/api/ThrowableCollector.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.util.List;
+
+public interface ThrowableCollector {
+
+	void collectError(Throwable t);
+
+	List<Throwable> errorsCollected();
```

Comment:
Let's return `List<AssertionError>` and rename this method to `assertionErrorsCollected()`

Reply:
See comments above.

Reply:
I've had a look at this and it's slightly complicated by the fact that `List<Throwable> errorsCollected()` was already a public method of `AbstractSoftAssertions`, so I can't just get rid of it. I'll add `List<AssertionError> assertionErrorsCollected()` and delegate from the existing method signature to the other.